fusestore/store/src/meta_service/state_machine.rs

```diff
@@ -289,13 +289,13 @@ impl StateMachine {
             Cmd::DropTable {
                 ref db_name,
                 ref table_name,
+                ref table_id,
+                if_exists: _,
             } => {
-                let db = self.databases.get(db_name);
-                let db = db.unwrap(); // unsafe?
+                let db = self.databases.get_mut(db_name).unwrap();
 
-                let table_id = db.tables.get(table_name);
-                let table_id = table_id.unwrap();
-                let prev = self.tables.get(table_id).cloned();
+                db.tables.remove(table_name);
+                let prev = self.tables.remove(table_id);
```

Comment:
Why do we need both table name and table id?
I suppose `db.tables[table_name]` is the table id.

Reply:
As I always use `db.tables.get(table_name)` to fetch table id, these two are used to avoid mutable(`db.tables.remove`) and immutable(`db.tables.get`) borrow at the same time :(

Reply:
Now I learned but still confuse why use `map[k]` syntax does not generate an immutable borrow❓

Reply:
> Now I learned but still confuse why use `map[k]` syntax does not generate an immutable borrow❓

Not very sure about what exactly your question is. Do you have a minimal snippet to show it?